server/src/main/java/org/apache/druid/server/coordinator/CuratorLoadQueuePeon.java

```diff
@@ -406,11 +409,11 @@ public void addCallback(LoadPeonCallback newCallback)
       }
     }
 
-    LoadPeonCallback[] getCallbacks()
+    List<LoadPeonCallback> snapshotCallbacks()
     {
       synchronized (callbacks) {
         // Return a copy so that callers get a consistent view
```

Comment:
I think the comment like the following would be better:
```
Return a copy to avoid a race condition, see
https://github.com/code-review-checklists/java-concurrency#unsafe-concurrent-iteration
```

Reply:
I think the comment I have makes more sense. While the possible inconsistency is because of the race condition, the comment delivers the fact that callers can expect a consistent view and need not worry about other threads concurrently modifying the returned collection.

Reply:
The risk is not just "inconsistency" (not very clear what does it mean), the risk is `NullPointerException`, `ArrayIndexOutOfBoundsException`.

When I stumbled upon this comment, I wondered what "inconsistency" is meant and had to look around the code to understand. This is what we want to avoid - comments and code that leaves more questions than answers to code readers.

The comment that I suggested might not be very clear as well. If you think so, this means that the comment should be even bigger, maybe combining information from our comments.

Reply:
I think @samarthjain's comment is fine. It's clear from context and from the comment that a copy of the List is being made so callers can iterate/read it without running into issues due to concurrent modification of that List by another thread. It is implied that the concurrency model of the class allows / expects this sort of usage.